app/src/main/kotlin/batect/execution/ContainerDependencyGraph.kt

```diff
@@ -72,13 +72,15 @@ data class ContainerDependencyGraph(val config: Configuration, val task: Task, v
 
             val dependencyNodes = resolveDependencies(dependencies, nodesAlreadyCreated, newPath)
             val command = containerCommandResolver.resolveCommand(container, task)
+            val entrypoint = container.entrypoint
```

Comment:
If you just want to add support for overriding the entrypoint on a container level (as opposed to at the task level) in this PR, you could get away with omitting this - anything that needs the entrypoint could take it from `container`. (The reason we have `command` here is so that we can store the command to use in the dependency graph node and so everything downstream of here doesn't have to know whether the command is coming from the task or the container.)